wallet/wallet-cli/src/main.rs

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{collections::VecDeque, path::PathBuf, sync::Arc};
+
+use clap::Parser;
+use common::chain::ChainConfig;
+use dialoguer::{theme::ColorfulTheme, History};
+use node_comm::{make_rpc_client, node_traits::NodeInterface};
+use utils::default_data_dir::PrepareDataDirError;
+
+use crate::config::Commands;
+
+mod config;
+
+const MAX_HISTORY: usize = 1000;
+
+#[derive(thiserror::Error, Debug)]
+pub enum WalletCliError {
+    #[error("RPC error: {0}")]
+    RpcError(node_comm::rpc_client::NodeRpcError),
+    #[error("Wallet error: {0}")]
+    WalletError(wallet::wallet::WalletError),
+    #[error("Console IO error: {0}")]
+    ConsoleIoError(std::io::Error),
+    #[error("Cookie file {0} read error: {1}")]
```

Comment:
`RPC authentication cookie-file {0} isn't found. Error: {1}`

Reply:
"isn't found" is not always accurate. There can also be permission errors. I changed it to this:
```
RPC authentication cookie-file {0} read error: {1}. Please make sure the node is started.
```

Reply:
The node does generate a cookie file no matter what, right? Because I remember in bitcoin you can choose whether you wanna generate a cookie file. I even remember that if you provide a username and password to the node, a cookie file won't be generated. What logic do we use?

If we have similar discrepancy, we may wanna do some test command first to ensure that the node is reachable. Besides, not being able to connect to the node isn't necessarily a crime. The connection can be lost at any moment and that should be OK.

Reply:
The node won't generate a cookie file if username/password is specified.